packages/flutter_tools/lib/src/android/gradle_utils.dart

```diff
@@ -198,7 +207,7 @@ String getGradleVersionForAndroidPlugin(Directory directory, Logger logger) {
     return templateDefaultGradleVersion;
   }
   final String buildFileContent = buildFile.readAsStringSync();
-  final Iterable<Match> pluginMatches = _androidGradlePluginRegExp.allMatches(buildFileContent);
+  final Iterable<Match> pluginMatches = _buildAndroidGradlePluginRegExp.allMatches(buildFileContent);
```

Comment:
The function below accepts both `RegExp`s, but here we only check for one - is this intentional?

Could this call `getAgpVersion` instead?

Reply:
good catch, and the function is easier to read.